_posts/en/newsletters/2020-07-29-newsletter.md

```diff
@@ -0,0 +1,153 @@
+    and the channel can be resolved using the appropriate unilateral
+    close protocol.
+
+## Selected Q&A from Bitcoin StackExchange
+
+*[Bitcoin StackExchange][bitcoin.se] is one of the first places Optech
+contributors look for answers to their questions---or when we have a
+few spare moments to help curious or confused users.  In
+this monthly feature, we highlight some of the top-voted questions and
+answers posted since our last update.*
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+FIXME:bitschmidty
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [C-Lightning 0.9.0rc3][C-Lightning 0.9.0] is a release candidate for
+  an upcoming major release.  It adds support for the updated `pay`
+  command and new `keysend` RPC described in [last week's
+  newsletter's][news107 notable] *notable code changes* section.  It also
+  includes several other notable changes and multiple bug fixes.
+
+- [Bitcoin Core 0.20.1rc1][Bitcoin Core 0.20.1] is a release candidate
+  for an upcoming maintenance release.  In addition to bug fixes and
+  some RPC behavior changes resulting from those fixes, the planned
+  release provides compatibility with recent versions of [HWI][topic
+  HWI] and its support for hardware wallet firmware released for the
+  [fee overpayment attack][].
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
+[Hardware Wallet Interface (HWI)][hwi], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #19473][] net: Add -networkactive option
+  FIXME:bitschmidty
+
+- [Bitcoin Core #18044][] adds support for [BIP339][] witness txid
+  (wtxid) transaction announcements and requests using the `inv` and
+  `getdata` P2P protocol messages.  This eliminates a potential
+  inefficiency related to duplicate announcements of segwit-spending
+  transactions, making it cleaner to implement improvements such as
+  segwit v1 transactions ([taproot][topic taproot]) and [package
+  relay][topic package relay].  See our [extended summary][news104
+  bip339] of BIP339 for details.
+
+    After being merged into Bitcoin Core's master branch, this PR was
+    [discussed][meeting xscript] during the weekly Bitcoin Core
+    Development Meeting in relation to taproot activation.  It was noted
+    that the version of taproot merged into Bitcoin Core (see PR
+    [#17977][Bitcoin Core #17977]) will likely be based on wtxid relay.
+    This will make it more complicated to backport taproot to the
```

Comment:
To summarise what I think the problem is:

- nodes that are unaware of segwit v1 (taproot) rules will reject taproot transactions that spend segwit v1 outputs since they're non-standard. See https://github.com/bitcoin/bitcoin/blob/b62fbf9e1c193464bca443076906b7ea47d56532/src/script/interpreter.cpp#L1552-L1554. (Note that we can send _to_ a segwit v1 address since https://github.com/bitcoin/bitcoin/pull/15846)
- if using txid relay, that node can't add the txid to the `recentRejects` filter, since the problem could just have been that the witness was mutated. See https://github.com/bitcoin/bitcoin/blob/0.20/src/net_processing.cpp#L2599-L2604 in v0.20.
- that means that taproot-aware nodes will announce taproot-spending transactions to their taproot-unaware peers, those peers will download and reject the transaction, but will then redownload the transaction (and rereject it) every time another peer announces it to them.
- with wtxid relay, this isn't a problem. The taproot-unaware node will store the wtxid in its recentRejects, and never try to  redownload it.
- therefore we should only announce taproot transactions using wtxids.

I'm not sure how difficult it actually is to backport wtxid relay. It seems to me that it shouldn't be too hard. v0.20 was only released a few weeks ago, and the wtxid PR was open long before that. I think we should check on how difficult it actually is before running with this.

Reply:
Agree with @jnewbery that it's not /that/ hard to add wtxidrelay to 0.20 -- it's just not a matter of cherry-picking the patch from master; the code will need to be adapted and re-reviewed. For most backports it /is/ just a mater of cherry-picking a patch, so this would just be "hard" in comparison to that, in my opinion.

The "only announce taproot transactions using wtxids" is an additional change needed (that's not done yet, afaik) to actually avoid 0.20 and earlier nodes from spamming themselves by re-requesting taproot spends that they don't understand by txid from every peer.

The drawback of releasing taproot in a 0.20.x release but only for block consensus but not tx relay is that it would both reduce the ability to actually use taproot when it activated (if you can't spend a taproot tx via p2p and get it reliably to the miners, taproot's not really usable), and, if people did use taproot, would reduce the effectiveness of compact block relay (since nodes wouldn't already have the taproot spending transactions). All of which is to say this approach still means waiting for 0.21.y before taproot's usable, so doesn't really speed anything up compared to just releasing taproot in 0.21.y.

I think it would be super-useful to expand on the explanation in the newsletter a bit; there isn't any other canonical reference of how taproot/wtxid interacts anywhere else to point people to apart from the not very explicit description in the wtxid PR description, and having one would be helpful :)